thirtyinch/src/main/java/net/grandcentrix/thirtyinch/internal/UiThreadExecutorAutoBinder.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (C) 2017 grandcentrix GmbH
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.grandcentrix.thirtyinch.internal;
+
+import net.grandcentrix.thirtyinch.TiLifecycleObserver;
+import net.grandcentrix.thirtyinch.TiPresenter;
+
+import java.util.concurrent.Executor;
+
+/**
+ * binds a ui thread executor to the presenter when this view is attached
+ */
+public class UiThreadExecutorAutoBinder implements TiLifecycleObserver {
+
+    private final TiPresenter mPresenter;
+
+    private final Executor mUiThreadExecutor;
+
+    public UiThreadExecutorAutoBinder(final TiPresenter presenter,
+            final Executor uiThreadExecutor) {
```

Comment:
When we expect a `UiThreadExecutor`, then change the `Executor` signature to it.
Otherwise we can put `any` Executor to it which isn't on the UiThread.